rundeckapp/grails-spa/packages/ui-trellis/src/app/pages/project-nodes-config/ProjectConfigurableForm.vue

```diff
@@ -0,0 +1,213 @@
+import { PropType } from "vue/dist/vue";
+
+interface Prop {
+  type: string;
+  defaultValue: any;
+  title: string;
+  required: boolean;
+  options: any;
+  allowed: string;
+  name: string;
+  desc: string;
+  staticTextDefaultValue: string;
+}
+
+interface NodeServiceConfigurableItemValues {
+  delay: string;
+  enabled: string;
+  firstLoadSynch: string;
+}
+interface HealthCheckConfigurableItemValues {
+  onstartup: string;
+  enabled: string;
+  period: string;
+  refreshHealthCheckCache: string;
+}
+interface ConfigurableItem {
+  name: string;
+  properties: [Prop];
+  propertiesMapping:
+    | NodeServiceConfigurableItemValues
+    | HealthCheckConfigurableItemValues;
+  values: NodeServiceConfigurableItemValues | HealthCheckConfigurableItemValues;
+}
+
+export default defineComponent({
+  name: "ProjectConfigurableForm",
+  components: {
+    PluginConfig,
+  },
+  data() {
+    return {
+      rundeckContext: {} as RundeckContext,
+      pluginConfig: {},
+      apiResponse: {},
+      isConfigSet: false,
+      cacheConfig: { config: {} },
+      massagedConfig: {},
+    };
+  },
+  props: {
+    category: {
+      type: String,
+      required: true,
+    },
+  },
+  methods: {
+    async loadConfig() {
+      try {
+        const config = await getProjectConfigurable(
+          window._rundeck.projectName,
+          this.category,
+        );
+        this.apiResponse = config.response["projectConfigurable"] as [
+          ConfigurableItem,
+        ];
+        let properties = [];
+        this.apiResponse.forEach((item: ConfigurableItem) => {
+          const itemName = item.name;
+          item.properties.forEach((prop: any) => {
+            const originalPropName = prop.name;
+            prop.name = itemName + "." + prop.name;
+            Object.entries(item.values).forEach(([key, value]) => {
+              if (originalPropName === key) {
+                this.massagedConfig[prop.name] = value;
+              }
+            });
+          });
+          properties.push(item.properties);
+        });
+        let resolvedProps = [];
+        properties.forEach((item: any) => {
+          item.forEach((prop: any) => {
+            prop.desc = prop.description;
+            resolvedProps.push(prop);
+          });
+        });
+        this.pluginConfig = {
+          type: "ResourceModelSource",
+          props: resolvedProps,
+        };
+        this.cacheConfig.config = this.massagedConfig;
+        this.isConfigSet = true;
+      } catch (err) {
+        this.notifyError("Error loading config: " + err.message);
+        console.error(err);
+      }
+    },
+    async saveConfig() {
+      const configMap = this.transformConfigToMap(this.cacheConfig.config);
+      this.cacheConfig.config = {
+        ...this.cacheConfig.config,
+        ...configMap,
+      };
+      this.cacheConfig.config = this.convertMapNumbersToStrings(
+        this.cacheConfig.config,
+      );
```

Comment:
since you're calling await this.loadConfig that will update cacheConfig, could you please change these lines to avoid mutating it twice?

```suggestion
      const configsToSave = this.convertMapNumbersToStrings(this.transformConfigToMap(this.finalExtraConfig.config)) 
```

and then update the setProjectConfigurable call to pass the configsToSave (or the name you prefer)